src/Shared/FileMatcher.cs

```diff
@@ -1672,13 +1672,15 @@ bool CompareIgnoreCase(char inputChar, char patternChar, int iIndex, int pIndex)
                 // We will mostly be comparing ASCII characters, check this first
                 if (inputChar < 128 && patternChar < 128)
                 {
-                    if (inputChar >= 'A' && inputChar <= 'Z' && patternChar >= 'a' && patternChar <= 'z')
+                    if (inputChar >= 'A' && inputChar <= 'Z')
                     {
-                        return inputChar + 32 == patternChar;
+                        // Evaluates to true if patternChar is equal to inputChar or to the lower-case counter-part.
+                        return inputChar == (patternChar & ~0x20);
```

Comment:
Could you explain how you came to this micro-optimization?

Reply:
The original code had to do 4 comparisons here: 2 to make sure that `inputChar` is in range and 2 to make sure that `patternChar` is in the corresponding range also. Then it compared them knowing that pattern, being lower-case, needs to have its ASCII value 32 more than input for them to match.

I simplified the condition to check only one side, in this case `inputChar`. And instead of adding/subtracting 32 I am setting/clearing the corresponding bit, making the check work regardless of what the value of `patternChar` is.

Example: If `inputChar` is `A` and `patternChar` is `A` then clearing the bit does nothing as upper case characters don't have it set so the result will be `true`, i.e. they are the same. If `inputChar` is `A` and `patternChar` is `a` then clearing the bit on `patternChar` turns it into `A` and again, the check works.

This line could equivalently be written as:

```C#
return (inputChar | 0x20) == patternChar;
```

A micro benchmark for `CompareIgnoreCase` showed ~10% perf improvement with this change in the common case where the input are letters from the English alphabet.

Reply:
Thank you, that's a great point! I have pushed your suggestion, just with `inputCharLower = ` as it's a bit easier to read than `x & ~y`.